src/value/borrowed.rs

```diff
@@ -158,20 +146,20 @@ impl<'val> Struct for BorrowedStruct<'val> {
         // convert (&k, &v[0..n]) -> (&k, &v[0])
         // flattens the fields_with_text_key HashMap and chains with fields_with_no_text_key
         // to return all fields with iterator
-        let all_fields_vec: Vec<&(BorrowedSymbolToken, BorrowedElement)> = self
-            .fields_with_text_key
-            .values()
-            .flat_map(|v| v)
-            .into_iter()
-            .chain(self.fields_with_no_text_key.iter())
-            .collect();
-        Box::new(all_fields_vec.into_iter().map(|kv| match &kv {
-            (k, v) => (k, v),
-        }))
+        Box::new(
+            self.text_fields
+                .values()
+                .flat_map(|v| v)
+                .into_iter()
+                .chain(self.no_text_fields.iter())
+                .map(|kv| match &kv {
+                    (k, v) => (k, v),
+                }),
```

Comment:
*Minor*: You should be able to do the de-structuring in the closure argument FWIW

```rust
map(|(k, v)| (k, v))
```

Just eliminates the extra `match` expression.